Sources/NIO/EventLoop.swift

```diff
@@ -498,30 +497,27 @@ internal final class SelectableEventLoop: EventLoop {
             // We need to ensure we process all tasks, even if a task added another task again
             while true {
                 // TODO: Better locking
-                tasksLock.lock()
-                if scheduledTasks.isEmpty {
-                    // Reset nextReadyTask to nil which means we will do a blocking select.
-                    nextReadyTask = nil
-                    tasksLock.unlock()
-                    break
-                }
-
-                // We only fetch the time one time as this may be expensive and is generally good enough as if we miss anything we will just do a non-blocking select again anyway.
-                let now = DispatchTime.now()
-
-                // Make a copy of the tasks so we can execute these while not holding the lock anymore
-                while tasksCopy.count < tasksCopy.capacity, let task = scheduledTasks.peek() {
-                    if task.readyIn(now) <= .nanoseconds(0) {
-                        _ = scheduledTasks.pop()
-                        tasksCopy.append(task.task)
+                tasksLock.withLockVoid {
+                    if !scheduledTasks.isEmpty {
+                        // We only fetch the time one time as this may be expensive and is generally good enough as if we miss anything we will just do a non-blocking select again anyway.
+                        let now = DispatchTime.now()
+
+                        // Make a copy of the tasks so we can execute these while not holding the lock anymore
+                        while tasksCopy.count < tasksCopy.capacity, let task = scheduledTasks.peek() {
+                            if task.readyIn(now) <= .nanoseconds(0) {
+                                _ = scheduledTasks.pop()
+                                tasksCopy.append(task.task)
+                            } else {
+                                nextReadyTask = task
+                                break
+                            }
+                        }
                     } else {
-                        nextReadyTask = task
-                        break
```

Comment:
The break is not needed anymore as `tasksCopy.isEmpty` will be true and so we break out the while loop.